Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -892,12 +885,306 @@ bool CSSPropertyParser::consumeColumns(bool important)
+        && isNumber(quad.left(), number, type);
+}
+
+static bool isNumber(const RectBase* quad, double number, CSSUnitType type)
+{
+    return quad && isNumber(*quad, number, type);
+}
+
+static bool isNumericQuad(const CSSValue& value, double number, CSSUnitType type)
+{
+    return isNumber(quadValue(value), number, type);
+}
+
+bool isInitialValueForLonghand(CSSPropertyID longhand, const CSSValue& value)
+{
+    if (value.isImplicitInitialValue())
+        return true;
+    switch (longhand) {
+    case CSSPropertyBackgroundSize:
+    case CSSPropertyMaskSize:
+        if (isValueIDPair(value, CSSValueAuto))
+            return true;
+        break;
+    case CSSPropertyBorderImageOutset:
+        if (isNumericQuad(value, 0, CSSUnitType::CSS_NUMBER))
+            return true;
+        break;
+    case CSSPropertyBorderImageRepeat:
+        if (isValueIDPair(value, CSSValueStretch))
+            return true;
+        break;
+    case CSSPropertyBorderImageSlice:
+        if (auto sliceValue = dynamicDowncast<CSSBorderImageSliceValue>(value)) {
+            if (!sliceValue->m_fill && isNumber(sliceValue->slices(), 100, CSSUnitType::CSS_PERCENTAGE))
+                return true;
+        }
+        break;
+    case CSSPropertyBorderImageWidth:
+        if (auto widthValue = dynamicDowncast<CSSBorderImageWidthValue>(value)) {
+            if (!widthValue->m_overridesBorderWidths && isNumber(widthValue->widths(), 1, CSSUnitType::CSS_NUMBER))
+                return true;
+        }
+        break;
+    default:
+        break;
+    }
+    return WTF::switchOn(initialValueForLonghand(longhand), [&](CSSValueID initialValue) {
+        return isValueID(value, initialValue);
+    }, [&](InitialNumericValue initialValue) {
+        return isNumber(value, initialValue.number, initialValue.type);
+    });
+}
+
+ASCIILiteral initialValueTextForLonghand(CSSPropertyID longhand)
+{
+    return WTF::switchOn(initialValueForLonghand(longhand), [](CSSValueID value) {
+        return nameLiteral(value);
+    }, [](InitialNumericValue initialValue) {
+        switch (initialValue.type) {
+        case CSSUnitType::CSS_NUMBER:
+            if (initialValue.number == 0.0)
+                return "0"_s;
+            if (initialValue.number == 1.0)
+                return "1"_s;
+            break;
+        case CSSUnitType::CSS_PERCENTAGE:
+            if (initialValue.number == 0.0)
+                return "0%"_s;
+            if (initialValue.number == 100.0)
+                return "100%"_s;
+            break;
+        case CSSUnitType::CSS_PX:
+            if (initialValue.number == 0.0)
+                return "0px"_s;
+            break;
+        case CSSUnitType::CSS_S:
+            if (initialValue.number == 0.0)
+                return "0s"_s;
+            break;
+        default:
+            break;
+        }
+        ASSERT_NOT_REACHED();
+        return ""_s;
+    }, [](std::monostate) {
+        ASSERT_NOT_REACHED();
+        return ""_s;
+    });
```

Comment:
I think you can remove this

Reply:
Yes, I’ll clean up after landing. Surprised it even compiles!